modules/apps/static/portal-configuration/portal-configuration-persistence-impl/src/main/java/com/liferay/portal/configuration/persistence/internal/ConfigurationPersistenceManager.java

```diff
@@ -406,11 +409,15 @@ private void _deleteFromDatabase(String pid) throws IOException {
 		}
 	}
 
-	private File _getCanonicalConfigFile(String fileName) throws IOException {
-		File configFile = new File(
-			PropsValues.MODULE_FRAMEWORK_CONFIGS_DIR, fileName);
+	private File _getConfigFile(String fileName) throws IOException {
+		Path configsDirPath = Paths.get(
+			PropsValues.MODULE_FRAMEWORK_CONFIGS_DIR);
+
+		configsDirPath = configsDirPath.toRealPath();
+
+		Path configFilePath = configsDirPath.resolve(fileName);
 
-		return configFile.getCanonicalFile();
+		return configFilePath.toFile();
 	}
```

Comment:
@ericyanLr, @tinatian.

This is just update for you.

Skip the test class, I referred to the [doc](https://www.maketecheasier.com/create-symbolic-links-windows10/) to create Symlink in windows and I created it successful. Refer to https://issues.liferay.com/browse/LPS-162706 description, but I can't reproduce the issue in windows. I can make sure Symlink took effect because the related configuration data can be created successful.

I asked @Emmaliu help me to reproduce the issue in Linux. And the issue can be reproduced in Linux on 7.3.x and master. Emma and I did the screenshot screen to debug the issue in Linux (Thanks @Emmaliu). I find the root issue is that **configFile.getCanonicalFile()** return the different results if fileName is Symlink in the Windows and Linux.

In Windows, it returns Symlink file which equals with felix.fileinstall.filename. So I can't reproduce the issue.
In Linux, it returns the real file (..data/abc.config in LPS-162706). And the issue happened.

Reply:
@ericyanLr @yuhai @tinatian So, if I understand it correct, here we shouldn't follow the symbol link to get the actual file name; instead, we should use the symbol link's name?

Reply:
> @ericyanLr @yuhai @tinatian So, if I understand it correct, here we shouldn't follow the symbol link to get the actual file name; instead, we should use the symbol link's name?

@dantewang, yes